tests/baselines/reference/jsDeclarationsOptionalTypeLiteralProps.js

```diff
@@ -0,0 +1,48 @@
+//// [foo.js]
+/**
+ * foo
+ *
+ * @public
+ * @param {object} opts
+ * @param {number} opts.a
+ * @param {number} [opts.b]
+ * @param {number} [opts.c]
+ * @returns {number}
+ */
+function foo({ a, b, c }) {
+    return a + b + c;
+}
+
+
+//// [foo.js]
+/**
+ * foo
+ *
+ * @public
+ * @param {object} opts
+ * @param {number} opts.a
+ * @param {number} [opts.b]
+ * @param {number} [opts.c]
+ * @returns {number}
+ */
+function foo({ a, b, c }) {
+    return a + b + c;
+}
+
+
+//// [foo.d.ts]
+/**
+ * foo
+ *
+ * @public
+ * @param {object} opts
+ * @param {number} opts.a
+ * @param {number} [opts.b]
+ * @param {number} [opts.c]
+ * @returns {number}
+ */
+declare function foo({ a, b, c }: {
+    a: number;
+    b?: number;
+    c?: number;

```

Comment:
These should be `number | undefined` for consistency with emit elsewhere.

Reply:
@andrewbranch Do we need to follow the same behavior as in the TypeScript example?


```ts
function foo(args: { x: number; y?: number }) {
}

//  .d.ts

declare function foo(args: {
    x: number;
    y?: number;
}): void;
```

[Playground](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABMOcAUBDATgcwM4BciA3ogB5FggC2ARgKZYDciAngPyU0NaIC+AShIAoPsKA)

Reply:
Or `| undefined` should be added only for JSDoc?

Reply:
Oh hang on—this test doesn’t have `--strictNullChecks`, my bad. I bet it already works for `--strictNullChecks`. I’m not sure whether the TS example you just gave is a bug. It probably reuses the type annotation node provided. But _generally_, when we serialize optional properties, we add the `| undefined` to the type annotation even if it wasn’t written by the user. But that would never show up without `--strictNullChecks`. Can we enable it for this test just to make sure we know how it behaves?

Reply:
@andrewbranch Sure! I've added [jsDeclarationsOptionalTypeLiteralProps2](https://github.com/microsoft/TypeScript/pull/42116/files#diff-208d846f4d46557112e779f86ad593ae2ee869042212e5e2ff564a6edf8164dc) with the `strict` option.